x-pack/filebeat/module/cisco/shared/ingest/asa-ftd-pipeline.yml

```diff
@@ -1,1253 +1,1272 @@
 description: "Pipeline for Cisco {< .internal_PREFIX >} logs"
 processors:
+  #
+  # Parse the syslog header
+  #
+  # This populates the host.hostname, process.name, timestamp and other fields
+  # from the header and stores the message contents in log.original.
+  - grok:
+      field: message
+      patterns:
+        - "(?:%{SYSLOG_HEADER})?\\s*%{GREEDYDATA:log.original}"
+      pattern_definitions:
+        SYSLOG_HEADER: "(?:%{SYSLOGFACILITY}\\s*)?(?:%{FTD_DATE:_temp_.raw_date}:?\\s+)?(?:%{PROCESS_HOST}|%{HOST_PROCESS})(?:{DATA})?%{SYSLOG_END}?"
```

Comment:
The lack of an optional `SYSLOG_END` was causing timestamps to get truncated in the test log. Furthermore, some logs have process prior to hostname, so added an alternative swapping of the hostname and process.